src/test/functional/functional-tests.js

```diff
@@ -94,7 +94,12 @@ describe('functional tests', function() {
   var _5mb = dataDir ? fs.readFileSync(dataDir + '/' + _5mbObjectName) : (new Buffer(5 * 1024 * 1024)).fill(0)
   var _5mbmd5 = crypto.createHash('md5').update(_5mb).digest('hex')
 
-  var customContentType = 'application/octet-stream'
+  var metaData = {
+    'Content-Type': 'text/html',
+    'Content-Language': 123,
+    'X-Amz-Meta-Testing': 1234,
+    'randomstuff': 5678
+  }
```

Comment:
is there a functional test case which verifies if the metadata is available in statObject? if not can you add one?

Reply:
I added one previously on line 449:
if (Object.keys(stat.metaData).length !== Object.keys(metaData).length) return done(new Error('content-type mismatch'))
Where I check to make sure they have the same number of keys.

Reply:
instead can you do this:
```
@@ -97,7 +97,7 @@ describe('functional tests', function() {
   var metaData = {
     'Content-Type': 'text/html',
     'Content-Language': 123,
-    'X-Amz-Meta-Testing': 1234,
+    'Testing': 1234,
     'randomstuff': 5678
   }
 
@@ -447,6 +447,9 @@ describe('functional tests', function() {
         if (e) return done(e)
         if (stat.size !== _100kb.length) return done(new Error('size mismatch'))
         if (Object.keys(stat.metaData).length !== Object.keys(metaData).length) return done(new Error('content-type mismatch'))
+       assert.equal(stat.metaData['Content-Type'], metaData['Content-Type'])
+       assert.equal(stat.metaData['Testing'], metaData['Testing'])
+       assert.equal(stat.metaData['randomstuff'], metaData['randomstuff'])
         etag = stat.etag
         modifiedDate = stat.modifiedDate
```

Reply:
done